diff --git a/build.gradle b/build.gradle
index 60d6e41..a039fd9 100644
--- a/build.gradle
+++ b/build.gradle
@@ -83,8 +83,10 @@ if(hasProperty('target') && target == 'android') {
     apply plugin: 'java'
     apply plugin: 'maven-publish'
 
-    sourceCompatibility = JavaVersion.VERSION_1_8
-    targetCompatibility = JavaVersion.VERSION_1_8
+    java {
+        sourceCompatibility = JavaVersion.VERSION_1_8
+        targetCompatibility = JavaVersion.VERSION_1_8
+    }
 
     publishing {
         publications {
@@ -96,7 +98,7 @@ if(hasProperty('target') && target == 'android') {
     }
 
     task execute(type:JavaExec) {
-       main = System.getProperty('mainClass')
+       getMainClass().set(System.getProperty('mainClass'))
        classpath = sourceSets.main.runtimeClasspath
     }
 }
diff --git a/pom.xml b/pom.xml
index ffa361b..3063c0b 100644
--- a/pom.xml
+++ b/pom.xml
@@ -31,6 +31,14 @@
         </developer>
     </developers>
 
+    <distributionManagement>
+        <repository>
+            <id>ossrh</id>
+            <name>Central Repository OSSRH</name>
+            <url>https://oss.sonatype.org/service/local/staging/deploy/maven2/</url>
+        </repository>
+    </distributionManagement>
+
     <build>
         <plugins>
            <plugin>
